Add nativefiledialog as a git submodule

DenisKolodin opened the pull request:
This PR uses **nativefiledialog** as submodule. I keep `build.rs`. But use `devel` branch. It's in active development. I'll change dependency to `master` when `devel` will be merged, because it's one still contains changes we need.

We can use `devel` or wait for `master`. Any option available. But PR is ready to check.

DenisKolodin commented:
My AppVeyor to check: https://ci.appveyor.com/project/DenisKolodin/nfd-rs
MinGW and VS versions were build successfully.

emoon commented on Cargo.toml:
```diff
@@ -1,7 +1,11 @@
+version = "0.0.3"
+authors = [
+	"Saurav Sachidanand <sauravsachidanand@gmail.com>",
+	"Daniel Collin <daniel@collin>",
```
I think .com is missing in my email. Not the end of the world but still :)

DenisKolodin replied:
Ok, I'll fix it )

DenisKolodin commented:
@emoon I've fixed your mail. Check, please, Is is right?

emoon commented:
Looks good :)

saurvs commented:
The devel branch has been merged into master.

DenisKolodin commented:
That's good news! I'll update this PR.

DenisKolodin commented:
I've updated this PR fully: made a new fork and added that chages again. I did it to keep submodules clean.

Building in progress: https://ci.appveyor.com/project/DenisKolodin/nfd-rs

DenisKolodin commented:
It's good if you can check it locally before merge. I'm checking for linux now...

DenisKolodin commented:
I checked PR with **Fedora 24** and **Windows 10** and examples. It works fine!
No other OS I can check, but I feel it will work on **Mac** too, because `nativefiledialog` hasn't major changes. Build system only, but we use own `build.rs`.

saurvs commented:
This PR works fine on OS X El Capitan too. Thanks for the contribution!

emoon commented:
👍
